### ISSUE TITLE
prettier@2.1.0 somehow broke storybook@6

### ISSUE THREAD

**Issue by Guria** [prettier/prettier#9039]
All details at https://github.com/storybookjs/storybook/issues/12217

Broken state:
[repro repo](https://github.com/Guria/cra-ts-storybook-issue) and [log from GitHub Actions run](https://github.com/Guria/cra-ts-storybook-issue/runs/1021991564?check_suite_focus=true#step:5:1047)

Fixed state:
https://github.com/Guria/cra-ts-storybook-issue/pull/2  [log from GitHub Actions run](https://github.com/Guria/cra-ts-storybook-issue/pull/2/checks?check_run_id=1022836548#step:5:1072)

I still haven't figured the way it affects storybook parsing, unfortunately.

**Comment by thorn0** [comment]
Storybooks is using Prettier in an [unexpected hacky way](https://github.com/storybookjs/storybook/blob/2bee5ddc43c16dcd1ba3e21a5fc00a643dde1f91/lib/source-loader/src/abstract-syntax-tree/parsers/parser-ts.js), so I believe this has to be fixed on the side of Storybooks, not in Prettier.

**Comment by thorn0** [comment]
The simplest workaround would be to use the `babel-ts` parser instead of `typescript`. I can't promise this won't break in the future though as `import ... from 'prettier/...'` means relying on Prettier's implementation details.

**Comment by Guria** [comment]
Yeah, thanks. I found that too. I proposed to pin to `prettier@2.0.x` as immediate action.
> I can't promise this won't break in the future though as `import ... from 'prettier/...'` means relying on Prettier's implementation details.

Actually imports like `prettier/...` is often considered to be public in many packages. If it were `prettier/lib/...` it could better tell people that it is internal implementantion detail. But much better to have eplicit note about what others could treat as public API in documentation I suppose.
